bin/metababel

```diff
@@ -78,6 +83,13 @@ class Babeltrace2Gen::BTTraceClass
       end
     end.flatten
   end
+  def find_event_class
+    @stream_classes.each.with_index do |s, index_stream_class|
+      s.event_classes.each.with_index do |e, index_event_class|
+        return [e, s.default_clock_class] if yield(e)
```

Comment:
What does `if yield(e)` even mean? Do you know @Kerilk ?

Reply:
`yield` represents the block passed to the `find_event_class` once called. The block should return a true or false value. When the block return true, the function `find_event_class` returns the `event` and the `clock`.

Reply:
Yep! Smart, never use it. 
We should rename to `find_all_event_classes` as you don't stop at the find one like `find` (https://apidock.com/ruby/Enumerable/find) 

Also please don't use `with_index` as you don't need it :)

Reply:
Agree with the changes, and we actually stop at the first occurrence as `find`.